Rubberduck.Core/UI/DockableToolwindowPresenter.cs

```diff
@@ -67,16 +67,15 @@ private IWindow CreateToolWindow(IDockableUserControl control)
                 Logger.Error(exception);
                 throw;
             }
-
-            // could cast to COM_IOleWindow for the first few, but AddUserControl is only on _DockableWindowHost
-            dynamic userControlHost = _userControlObject;
+            
             toolWindow.IsVisible = true; //window resizing doesn't work without this
-
             EnsureMinimumWindowSize(toolWindow);
-
             toolWindow.IsVisible = _settings != null && _settings.IsWindowVisible(this);
 
-            userControlHost.AddUserControl(control as UserControl, new IntPtr(_vbe.MainWindow.HWnd));
+            // currently we always inject _DockableToolWindowHost from Rubberduck.Main.
+            // that method is not exposed in any of the interfaces we know, though, so we need to invoke it blindly
+            ((dynamic)_userControlObject).AddUserControl(control as UserControl, new IntPtr(_vbe.MainWindow.HWnd));
```

Comment:
While the comment is much better than the original, I'm left wondering why aren't we exposing it properly in the first place and thus not need to do the `dynamic` gymnastics?

Reply:
The problem is that currently all com visible non-API classes are in `Rubberduck.Main`. Because this assembly contains the IoC container it references everything else and, thus, cannot be referenced by any other assembly.

The alternative to this code here, of which I am not even fully sure that it works as intended, is to put the com visible non-API classes into a separate assembly, leaving only the IoC container in `Rubberduck.Main`, and to do some reflection voodoo to get the IoC container in the entry point `Extension`, which also has to be com visible.

Reply:
I probably would just update the comment to include that explanation, and leave it as a TODO to fix in a future PR. It's better that we first answer the question about how we will handle the COM registration going forward before we mess with that piece of code.